src/main/java/de/doubleslash/keeptime/view/ReportController.java

```diff
@@ -211,74 +196,62 @@ protected void updateItem(TableRow workItem, boolean empty) {
    }
+
    private void updateReport(final LocalDate dateToShow) {
+      Report report = new Report(dateToShow, model, controller);
+
       this.currentReportDate = dateToShow;
       rootItem.getChildren().clear();
       reportRoot.requestFocus();
 
-      this.currentDayLabel.setText(DateFormatter.toDayDateString(this.currentReportDate));
-      final List<Work> currentWorkItems = model.getWorkRepository()
-                                               .findByStartDateOrderByStartTimeAsc(this.currentReportDate);
-
-      colorTimeLine.update(currentWorkItems, controller.calcSeconds(currentWorkItems));
-
-      final SortedSet<Project> workedProjectsSet = currentWorkItems.stream()
-                                                                   .map(Work::getProject)
-                                                                   .collect(Collectors.toCollection(() -> new TreeSet<>(
-                                                                         Comparator.comparing(Project::getIndex))));
-
-      long currentWorkSeconds = 0;
-      long currentSeconds = 0;
+      this.currentDayLabel.setText(report.getDateString());
 
-      for (final Project project : workedProjectsSet) {
-         final List<Work> onlyCurrentProjectWork = currentWorkItems.stream()
-                                                                   .filter(w -> w.getProject() == project)
-                                                                   .collect(Collectors.toList());
+      colorTimeLine.update(report.getWorkItems(), report.getWorkItemsSeconds());
 
-         final long projectWorkSeconds = controller.calcSeconds(onlyCurrentProjectWork);
-
-         currentSeconds += projectWorkSeconds;
-         if (project.isWork()) {
-            currentWorkSeconds += projectWorkSeconds;
-         }
+      for (Project project : report.getWorkedProjectsSet()) {
+         final List<Work> projectWorks = report.getWorkItems()
+                                               .stream()
+                                               .filter(w -> w.getProject().getId() == project.getId())
+                                               .toList();
 
          final HBox projectButtonBox = new HBox();
-         projectButtonBox.getChildren().add(createCopyProjectButton(onlyCurrentProjectWork));
-
+         projectButtonBox.getChildren().add(createCopyProjectButton(projectWorks));
          final TreeItem<TableRow> projectRow = new TreeItem<>(
-               new ProjectTableRow(project, projectWorkSeconds, projectButtonBox));
+               new ProjectTableRow(project, report.getProjectWorkSecondsMap().get(project.getId()), projectButtonBox));
 
-         for (final Work w : onlyCurrentProjectWork) {
+         for (final Work work : projectWorks) {
             final HBox workButtonBox = new HBox(5.0);
-            if(w.getId()==model.activeWorkItem.get().getId()){
+
+            if (work.getId() == model.activeWorkItem.get().getId() || work.getId() == 0) {
```

Comment:
reminder for me: I do not like having this check still in the view-controller. Idea was that view just stupidly can show what the "view-model" presents it. no special logic needed.
this is why we did not merge it yet - as we now have logic in 2 places (Report.java and this class)